From 6cc4b18ad5e3830b33c3c48c896c05c7be0cee75 Mon Sep 17 00:00:00 2001
From: Adrian Alonso <adrian.alonso@nxp.com>
Date: Mon, 19 Sep 2016 14:33:58 -0500
Subject: [PATCH 1161/5242] MLK-13243: arm: imx6q: busfreq: lpddr2 fix system
 clocks audio mode

commit  417ede3028821f23d40fea52471057cb637a2029 from
https://source.codeaurora.org/external/imx/linux-imx.git

Fix system clock topology used by lpddr2 for audio mode
Keep pll2_pfd2 as clock root for periph_pre_clk to match
lpddr2_freq_imx6q.S switching mechanism.

(Rework from commit id 427b1b6d628827ca83887b92c8331a261a254151)

Signed-off-by: Adrian Alonso <adrian.alonso@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/busfreq-imx.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/busfreq-imx.c b/arch/arm/mach-imx/busfreq-imx.c
index 959ceb7..5e9ce18 100644
--- a/arch/arm/mach-imx/busfreq-imx.c
+++ b/arch/arm/mach-imx/busfreq-imx.c
@@ -267,7 +267,10 @@ static void enter_lpm_imx6_smp(void)
 			update_lpddr2_freq_smp(HIGH_AUDIO_CLK);
 		/* Make sure periph clk's parent also got updated */
 		clk_set_parent(periph_clk2_sel_clk, pll3_clk);
-		clk_set_parent(periph_pre_clk, pll2_200_clk);
+		if (ddr_type == MMDC_MDMISC_DDR_TYPE_DDR3)
+			clk_set_parent(periph_pre_clk, pll2_200_clk);
+		else if (ddr_type == MMDC_MDMISC_DDR_TYPE_LPDDR2)
+			clk_set_parent(periph_pre_clk, pll2_400_clk);
 		clk_set_parent(periph_clk, periph_pre_clk);
 		audio_bus_freq_mode = 1;
 		low_bus_freq_mode = 0;
-- 
1.7.9.5

